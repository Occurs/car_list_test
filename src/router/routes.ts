const routes = {
  carsList: '/',
  carCard: '/car/:stockNumber',
  favorites: '/favorites'
};

export default routes;