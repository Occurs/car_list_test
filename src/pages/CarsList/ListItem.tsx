import React from 'react';
import Link from '@material-ui/core/Link';
import { NavLink, generatePath } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import { colors } from 'styles/variables';
import routes from 'router/routes';
import { ICar } from 'types/types';

type IListItemProps = {
  car: ICar,
}

const useStyles = makeStyles({
  image: {
    width: 'inherit',
    height: 'inherit',
  },
});

const ListItem = ({ car }: IListItemProps) => {
  const { stockNumber, pictureUrl, modelName, manufacturerName, fuelType, color, mileage } = car;
  const classes = useStyles();
  const colorCapitalized = color.charAt(0).toUpperCase() + color.slice(1);
  return (
    <Box display='flex' border={`2px solid ${colors.secondary}`} padding='12px' height='90px' marginTop='12px' width='100%'>
      <Box component='div' bgcolor={colors.secondary} marginRight='12px' height='100%' width='85px'>
        <img alt={`${stockNumber}`} src={pictureUrl} className={classes.image} />
      </Box>
      <Box component='div' marginLeft='12px' height='100%' display='flex' alignItems='flex-start' flexDirection='column' justifyContent='space-between'>
        <Typography>
          <Box component='span' fontSize='18px' fontWeight='bold'>
            {`${modelName} ${manufacturerName}`}
          </Box>
        </Typography>
        <Typography>
          <Box component='span' fontSize='12px'>
            {`Stock # ${stockNumber} - ${mileage.number} ${mileage.unit.toUpperCase()} - ${fuelType} - ${colorCapitalized}`}
          </Box>
        </Typography>
        <Link component={NavLink} to={generatePath(routes.carCard, { stockNumber: car.stockNumber })}>
          <Box fontSize='12px' color={colors.primary} component='span'>View details</Box>
        </Link>
      </Box>
    </Box>
  );
};

export default ListItem;
