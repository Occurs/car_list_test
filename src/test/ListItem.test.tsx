import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import ListItem from "pages/CarsList/ListItem";
import routes from "router/routes";

const props = {
  car: {
    stockNumber: 41400,
    manufacturerName: "Fiat",
    modelName: "Marea",
    mileage: {
      number: 100141,
      unit: "km",
    },
    fuelType: "Diesel",
    color: "white",
    pictureUrl: "",
  },
};

const renderComponent = () =>
  render(
    <MemoryRouter initialEntries={[routes.carsList]}>
      <ListItem {...props} />
    </MemoryRouter>
  );

describe("Car ListItem Component", () => {
  test("snapshot", () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={[routes.carsList]}>
        <ListItem {...props} />
      </MemoryRouter>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("render", async () => {
    const { getByText } = renderComponent();
    getByText("Fiat Marea");
    getByText("Stock # 41400 - 100141 KM - Diesel - White");
  });
});
