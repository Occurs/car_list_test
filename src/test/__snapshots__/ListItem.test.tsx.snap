// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car ListItem Component snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    <img
      alt="41400"
      className="makeStyles-image-1"
      src=""
    />
  </div>
  <div
    className="MuiBox-root MuiBox-root-4"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <span
        className="MuiBox-root MuiBox-root-5"
      >
        Fiat Marea
      </span>
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <span
        className="MuiBox-root MuiBox-root-6"
      >
        Stock # 41400 - 100141 KM - Diesel - White
      </span>
    </p>
    <a
      aria-current={null}
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="/car/41400"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      <span
        className="MuiBox-root MuiBox-root-7"
      >
        View details
      </span>
    </a>
  </div>
</div>
`;
